src/backend/distributed/deparser/deparse_owned_stmts.c

```diff
@@ -0,0 +1,88 @@
+/*-------------------------------------------------------------------------
+ *
+ * deparse_owned_stmts.c
+ *	  Functions to turn all Statement structures related to owned back
+ *    into sql.
+ *
+ * Copyright (c), Citus Data, Inc.
+ *
+ *-------------------------------------------------------------------------
+ */
+
+#include "postgres.h"
+
+#include "pg_version_compat.h"
+
+#include "distributed/citus_ruleutils.h"
+#include "distributed/deparser.h"
+#include "lib/stringinfo.h"
+#include "nodes/parsenodes.h"
+#include "utils/builtins.h"
+
+static void AppendDropOwnedStmt(StringInfo buf, DropOwnedStmt *stmt);
+static void AppendRoleList(StringInfo buf, List *roleList);
+
+/*
+ * DeparseDropOwnedStmt builds and returns a string representing of the
+ * DropOwnedStmt for application on a remote server.
+ */
+char *
+DeparseDropOwnedStmt(Node *node)
+{
+	DropOwnedStmt *stmt = castNode(DropOwnedStmt, node);
+
+	StringInfoData buf = { 0 };
+	initStringInfo(&buf);
+
+	AppendDropOwnedStmt(&buf, stmt);
+
+	return buf.data;
+}
+
+
+/*
+ * AppendDropOwnedStmt generates the string representation of the
+ * DropOwnedStmt and appends it to the buffer.
+ */
+static void
+AppendDropOwnedStmt(StringInfo buf, DropOwnedStmt *stmt)
+{
+	appendStringInfo(buf, "DROP OWNED BY ");
+
+	AppendRoleList(buf, stmt->roles);
+
+	if (stmt->behavior == DROP_RESTRICT)
+	{
+		appendStringInfo(buf, " RESTRICT");
+	}
+	else if (stmt->behavior == DROP_CASCADE)
+	{
+		appendStringInfo(buf, " CASCADE");
+	}
+}
+
+static void
+AppendRoleList(StringInfo buf, List *roleList)
+{
+	ListCell *cell = NULL;
+	foreach(cell, roleList)
+	{
+		Node *roleNode = (Node *) lfirst(cell);
+		Assert(IsA(roleNode, RoleSpec) || IsA(roleNode, AccessPriv));
+		char const *rolename = NULL;
+		if (IsA(roleNode, RoleSpec))
+		{
+			rolename = RoleSpecString((RoleSpec *) roleNode, true);
+		}
+		if (IsA(roleNode, AccessPriv))
```

Comment:
Can you really have this? Postgres code seems to deal with only RoleSpec: https://github.com/postgres/postgres/blob/3d14e171e9e2236139e8976f3309a588bcc8683b/src/backend/commands/user.c#L1467